resources/benchmark-runner.mjs

```diff
@@ -256,16 +168,41 @@ export class BenchmarkRunner {
                     if (this._client && this._client.didRunTest)
                         this._client.didRunTest(suite, test);
 
-                    state.next();
-                    resolve(state);
+                    resolve();
                 });
             }, 0);
         });
     }
 
-    _finalize() {
-        this._removeFrame();
+    // This function ought be as simple as possible. Don't even use Promise.
+    _runTest(suite, test, page, callback)
```

Comment:
Can we keep this function above so that diff will be smaller??

Reply:
DONE